security.sh

```diff
@@ -1,10 +1,19 @@
 #!/usr/bin/env bash
 echo ${TEST_URL}
-zap-api-scan.py -t ${TEST_URL}/v2/api-docs -f openapi -S -d -u ${SecurityRules} -P 1001 -l FAIL
-curl --fail http://0.0.0.0:1001/OTHER/core/other/jsonreport/?formMethod=GET --output report.json
 export LC_ALL=C.UTF-8
 export LANG=C.UTF-8
-zap-cli --zap-url http://0.0.0.0 -p 1001 report -o /zap/api-report.html -f html
-zap-cli --zap-url http://0.0.0.0 -p 1001 alerts -l High --exit-code False
-cp /zap/api-report.html functional-output/
-cat zap.out
\ No newline at end of file
+export PYTHONDONTWRITEBYTECODE=1
+
+echo "Run ZAP scan and generate reports"
+zap-api-scan.py -t ${URL_FOR_SECURITY_SCAN}/v2/api-docs -f openapi -S -d -u ${SECURITY_RULES} -P 1001 -l FAIL --hook=zap_hooks.py -J report.json -r api-report.html
+echo "Print alerts"
+zap-cli --zap-url http://0.0.0.0 -p 1001 alerts -l Informational --exit-code False
+
+echo "LC_ALL: ${LC_ALL}"
+echo "LANG: ${LANG}"
+echo "PYTHONDONTWRITEBYTECODE: ${PYTHONDONTWRITEBYTECODE}"
+echo "Print zap.out logs:"
+cat zap.out
+echo "Copy artifacts for archiving"
+cp zap.out functional-output/
```

Comment:
Might need this line after line 18
cp report.json functional-output/

as report.json is passed as param in line 8.. probably need to copy the output of it

Reply:
done